superset-frontend/src/common/hooks/apiResources/dashboards.ts

```diff
@@ -25,8 +25,9 @@ export const useDashboard = (idOrSlug: string | number) =>
     useApiV1Resource<Dashboard>(`/api/v1/dashboard/${idOrSlug}`),
     dashboard => ({
       ...dashboard,
-      metadata: JSON.parse(dashboard.json_metadata),
-      position_data: JSON.parse(dashboard.position_json),
+      metadata: dashboard.json_metadata && JSON.parse(dashboard.json_metadata),
```

Comment:
should this be initialized as a empty object if dashboard comes back as undefined.

Reply:
I assume you mean if the metadata is undefined (if the dashboard is undefined it will be a 404). I don't think it should be initialized like that, null is different from an empty object.

This is not actually new behavior, I just had to write this guard to appease the type system since I corrected the types of some dashboard fields to be nullable.